nima/tests/integration/webserver/webserver/src/test/java/io/helidon/nima/tests/integration/server/ErrorHandlingWithOutputStreamTest.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @SetUpRoute
+    static void router(HttpRouting.Builder router) {
+        router.error(CustomException.class, new CustomRoutingHandler())
+                .get("get-outputStream", (req, res) -> {
+                    res.outputStream();
+                    throw new CustomException();
+                })
+                .get("get-outputStream-writeOnceThenError", (req, res) -> {
+                    OutputStream os = res.outputStream();
+                    os.write("writeOnceOnly".getBytes(StandardCharsets.UTF_8));
+                    throw new CustomException();
+                })
+                .get("get-outputStream-writeTwiceThenError", (req, res) -> {
+                    OutputStream os = res.outputStream();
+                    os.write("writeOnce".getBytes(StandardCharsets.UTF_8));
+                    os.write("|writeTwice".getBytes(StandardCharsets.UTF_8));
+                    throw new CustomException();
+                })
+                .get("get-outputStream-writeFlushThenError", (req, res) -> {
+                    OutputStream os = res.outputStream();
+                    os.write("writeOnce".getBytes(StandardCharsets.UTF_8));
+                    os.flush();
+                    throw new CustomException();
+                })
+                .get((req, res) -> res.send("ok"));
+    }
+
+    @Test
+    void testOk() {
+        var response = client.get().request();
+
+        assertThat(response.status(), is(Http.Status.OK_200));
+        assertThat(response.entity().as(String.class), is("ok"));
+    }
+
+    @Test
+    void testGetOutputStreamThenError_expect_CustomErrorHandlerMessage() {
+        var response = client.get("/get-outputStream").request();
+
+        assertThat(response.status(), is(Http.Status.I_AM_A_TEAPOT_418));
+        assertThat(response.entity().as(String.class), is("CustomErrorContent"));
+    }
+
+    @Test
+    void testGetOutputStreamWriteOnceThenError_expect_CustomErrorHandlerMessage() {
+        var response = client.get("/get-outputStream-writeOnceThenError").request();
+
+        assertThat(response.status(), is(Http.Status.I_AM_A_TEAPOT_418));
+        assertThat(response.entity().as(String.class), is("CustomErrorContent"));
+    }
+
+    @Test
+    void testGetOutputStreamWriteTwiceThenError_expect_CloseConnectionExceptionAndContentFromMainHandler() {
+        Headers headers;
+        try (Http1ClientResponse response = client.method(Http.Method.GET)
+                .uri("/get-outputStream-writeTwiceThenError")
+                .request()) {
+            assertThat(response.status(), is(Http.Status.OK_200));
+            assertThrows(DataReader.InsufficientDataAvailableException.class, () -> response.entity().as(String.class));
+            headers = response.headers();
+        }
+        // FIXME: It this supposed to be a (trailing) header for connection close ??
+        // assertThat(headers, hasHeader(Http.HeaderValues.CONNECTION_CLOSE));
```

Comment:
As I see it this FIXME should be removed. We should not expect the Close Connection header here in this case. We expect the chunked data to not be terminated correctly and hence the InsufficientDataAvailableException and that should be correct and sufficient.

I should remove this FIXME comment from here.